lib/karafka/params/params_batch.rb

```diff
@@ -22,19 +22,24 @@ def initialize(messages_batch, topic_parser)
       # @note Invocation of this method will cause loading and parsing each param after another.
       #   If you want to get access without parsing, please access params_batch directly
       def each
-        @params_batch.each { |param| yield(param.retrieve!) }
+        @params_batch.each { |param| yield(param.parse!) }
       end
 
       # @return [Array<Karafka::Params::Params>] returns all the params in a loaded state, so they
       #   can be used for batch insert, etc. Without invoking all, up until first use, they won't
       #   be parsed
-      def parsed
+      def parse!
         each(&:itself)
       end
 
-      # @return [Karafka::Params::Params] last element after the unparsing process
+      # @return [Karafka::Params::Params] first element after the parsing process
```

Comment:
I've added this method to mimic the `#last` behaviour that returns a parsed instance. enumerator already had this method but would return unparsed which would cause inconsistency.

Reply:
Maybe add this to the CHANGELOG or is this only for internal use?